Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -249,11 +250,10 @@ CGImageRef CGImageCreate(size_t width,
                          const float* decode,
                          bool shouldInterpolate,
                          CGColorRenderingIntent intent) {
-    RETURN_NULL_IF(((provider == nullptr) || ![(NSObject*)provider isKindOfClass:[NSData class]]) || (colorSpace == nullptr));
+    RETURN_NULL_IF(provider == nullptr || colorSpace == nullptr);
 
-    NSData* dataProvider = (__bridge NSData*)provider;
-
-    unsigned char* data = (unsigned char*)[dataProvider bytes];
+    woc::unique_cf<CFDataRef> providerData{ CGDataProviderCopyData(provider) };
```

Comment:
use _CGDataProviderGetData ? rather than taking a copy?

---
In reply to: [95705695](https://github.com/Microsoft/WinObjC/pull/1687#discussion_r95705695) [](ancestors = 95705695)